src/upgrade/upgrade_scripts/5.14.0/upgrade_bucket_policy.js

```diff
@@ -0,0 +1,53 @@
+/* Copyright (C) 2023 NooBaa */
+"use strict";
+
+const util = require('util');
+const _ = require('lodash');
+const { OP_NAME_TO_ACTION } = require('../../../endpoint/s3/s3_utils');
+
+async function run({ dbg, system_store, system_server }) {
+
+    try {
+        dbg.log0('Starting bucket policy...');
+        const buckets = [];
+        const all_op_names = _.compact(_.flatMap(OP_NAME_TO_ACTION, action => [action.regular, action.versioned]));
+        for (const bucket of system_store.data.buckets) {
+            //Do not update if there are no bucket policy.
+            if (!bucket.s3_policy) return;
+
+            const new_policy = {};
+            if (bucket.s3_policy.version) new_policy.Version = bucket.s3_policy.version;
+
+            new_policy.Statement = bucket.s3_policy.statement.map(statement => ({
+                Effect: statement.effect === 'allow' ? 'Allow' : 'Deny',
+                Action: statement.action.map(atatement_action =>
+                    all_op_names.find(action => action.toLowerCase() === atatement_action)),
+                Principal: { AWS: statement.principal },
+                Resource: statement.resource,
+                Sid: statement.sid
+            }));
+
+            buckets.push({
+                _id: bucket._id,
+                s3_policy: new_policy,
+            });
+        }
+
+        if (buckets.length > 0) {
+            dbg.log0(`Replacing bucket policy rules to the new API structure for buckets: ${buckets.map(bucket => util.inspect(bucket)).join(', ')}`);
+            await system_store.make_changes({ update: { buckets } });
+        } else {
+            dbg.log0('Upgrading buckets policy: no upgrade needed...');
+        }
+
```

Comment:
we wanted to start creating tests for upgrades, I think it's a great start for creating upgrade tests
(I guess no need to create a real upgrade flow, just put in the db old bucket policies, call run() function of this update script, and then validate that the bucket policies were updated correctly)